features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java

```diff
@@ -110,26 +110,12 @@
 public class Deployer {
 
     public interface DeployCallback {
-
         void print(String message, boolean verbose);
-
         void saveState(State state);
         void persistResolveRequest(DeploymentRequest request) throws IOException;
         void installFeature(Feature feature) throws IOException, InvalidSyntaxException;
```

Comment:
I think it would make sense to move the `installFeature` too, as the `DeployCallback` would then only contain state and listeners management.